src/sentry/api/bases/organization.py

```diff
@@ -317,20 +317,25 @@ def has_release_permission(self, request, organization, release):
         on the projects to which the release is attached?
 
         If the given request has an actor (user or ApiKey), cache the results
-        for a minute on the unique combination of actor,org,release.
+        for a minute on the unique combination of actor,org,release, and project
+        ids.
         """
         actor_id = None
         has_perms = None
+        projects = self.get_projects(request, organization)
```

Comment:
Feels like this defeats the purpose of the caching? Previously we weren't making any db queries unless we had a cache miss.

Reply:
Hmm, I still think its worth it cause the Releases query is cached. But another alternative is to disable the cache 🤔

Reply:
I'm not sure why we actually made this optimization, @JTCunning was this a hot path you were fixing?

Reply:
@wedamija Went with using the ids from the request directly, even though they aren't validated. We still validate on cache set.

Messaged JT in slack, and he doesn't remember completely, but if he had to guess it was being hit by the event processing pipeline.

Reply:
Look good, yeah I was concerned about it somehow being involved in event processing since I remember @JTCunning doing a bunch of optimization around that late last year.